app/gui2/src/stores/awareness.ts

```diff
@@ -0,0 +1,81 @@
+import { Vec2 } from '@/util/vec2'
+import type { StackItem } from 'shared/languageServerTypes'
+import { reactive } from 'vue'
+import { Awareness as YjsAwareness } from 'y-protocols/awareness'
+import * as Y from 'yjs'
+
+// === Public types ===
+
+export type FileName = string
+
+export interface UploadingFile {
+  sizePercentage: number
+  stackItem: StackItem
+  position: Vec2
+}
+
+// === Awareness wrapper ===
+
+/**
+ * A thin wrapper around `Awareness` from `yjs`, providing helper methods for Enso IDE-specific state.
+ */
+export class Awareness {
+  public internal: YjsAwareness
+  uploadingFiles: Map<ClientId, Uploads>
```

Comment:
As this class is used in the project store, I again can’t make this field and some methods `private`. Similar issues were faced when implementing reactive db, `pinia` does not like private fields.

Reply:
just a heads up that fields are `public` by default. are we using `public` for documentation purposes then?

Reply:
I want to mark them as `private`, but typecheck fails in this case. We saw similar thing with @Frizi while working on reactive db, and we didn’t find an easy solution then.

Reply:
I think I found a solution for that. You can mark the fields as private as usual, and then use `markRaw` when attaching the class instance to the store. That would be `awareness: markRaw(awareness)` in the project store in this case.

Reply:
Actually it does not work, it breaks reactivity of awareness, which I need.

Reply:
It should work. I think the issue we discussed (the `filter` comparison) is actually a bug, and that change simply made it clearly visible. See a comment below.